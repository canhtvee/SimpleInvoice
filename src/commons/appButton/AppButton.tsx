import React from 'react';
import {
  ActivityIndicator,
  StyleSheet,
  Text,
  StyleProp,
  TextStyle,
  Pressable,
  PressableProps,
  ViewStyle,
} from 'react-native';

import {Colors, Sizes} from '../../utils';

export interface AppButtonProps extends PressableProps {
  title?: string;
  titleStyle?: StyleProp<TextStyle>;
  isLoading?: boolean;
}

export function AppButtonNormal({
  title,
  titleStyle,
  style,
  isLoading,
  disabled,
  onPress,
  ...touchProps
}: AppButtonProps) {
  let content;
  if (isLoading) {
    content = <ActivityIndicator color={Colors.onPrimary} size={Sizes.icon} />;
  } else {
    content = (
      <Text style={[styles.title, {color: Colors.onPrimary}, titleStyle]}>
        {title}
      </Text>
    );
  }

  return (
    <Pressable
      hitSlop={Sizes.paddinglx}
      disabled={disabled || isLoading}
      style={({pressed}) => {
        return StyleSheet.flatten([
          styles.primaryButton,
          {
            backgroundColor: Colors.primary,
            opacity: pressed ? 0.6 : 1,
          },
          style,
        ]) as ViewStyle;
      }}
      onPress={onPress}
      {...touchProps}>
      {content}
    </Pressable>
  );
}

const styles = StyleSheet.create({
  primaryButton: {
    justifyContent: 'center',
    alignItems: 'center',
    height: Sizes.wpx(40),
    paddingHorizontal: Sizes.padding * 2,
    borderRadius: Sizes.borderRadius,
  },

  title: {fontSize: Sizes.button, fontWeight: '600'},
});
