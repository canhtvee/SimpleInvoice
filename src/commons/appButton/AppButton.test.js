import React from 'react';

import {cleanup, fireEvent, render} from '@testing-library/react-native';

import {AppButton} from './AppButton';

/**
 * Run test command
 * yarn test -u -t="AppButton"
 */

afterEach(cleanup);

describe('AppButton', () => {
  test('should render correctly', () => {
    const tree = render(
      <AppButton title="AppButton" containerStyle={{padding: 50}} />,
    );

    expect(tree).toMatchSnapshot();
  });

  test('should invoke onPressMock if pressed ', () => {
    const onPressMock = jest.fn();

    const tree = render(
      <AppButton
        title="AppButton"
        containerStyle={{padding: 50}}
        onPress={onPressMock}
      />,
    );

    fireEvent.press(tree.getByText('AppButton'));
    expect(onPressMock).toHaveBeenCalled();
  });
});
