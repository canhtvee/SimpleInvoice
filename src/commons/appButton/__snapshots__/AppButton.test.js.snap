// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppButton should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  containerStyle={
    {
      "padding": 50,
    }
  }
  focusable={true}
  hitSlop={16}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#2196f3",
      "borderRadius": 6,
      "height": 79,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 47,
    }
  }
>
  <Text
    style={
      [
        {
          "fontSize": 29.5,
          "fontWeight": "600",
        },
        {
          "color": "#FFFFFF",
        },
        undefined,
        undefined,
      ]
    }
  >
    AppButton
  </Text>
</View>
`;
